src/main/java/gov/nasa/pds/tools/validate/rule/pds4/TableDataContentValidationRule.java

```diff
@@ -426,14 +447,23 @@ public void validateTableDataContents() throws MalformedURLException,
           } else {  // We have either a character or delimited table
             boolean manuallyParseRecord = false;
             String line = reader.readNextLine();
+            int lineNumber = 0;
             while (line != null) {
               progressCounter();
+              lineNumber += 1;
               
               if (!line.endsWith("\r\n")) {
```

Comment:
@qchaupds per #292, I think the logic can be much simpler here. most importantly, per this ticket, the check for line ending must be keyed off of the value in the `record_delimiter` attribute. here is some pseudocode for a simple check:
```
// Very similar to the original logic in this method, 
// except we first read the value of the <record_delimiter> attribute
if record_delimiter.value() == "Carriage-Return Line-Feed" AND !line.endsWith("\r\n"):
    throw ERROR
// Next, we need to do an else if because CRLF ends in a new line so we need to do some additional checks
else if record_delimiter.value() == "Line-Feed":
    if (line.endsWith("\r\n") OR !line.endsWith("\n"):
        throw an ERROR
```

Per the sample data, here is an example  of where you can find `record_delimiter`:
```
  <File_Area_Observational>
    <File>
      <file_name>minimal_test_product_crlf.tab</file_name>
      <file_size unit="byte">240</file_size>
      <md5_checksum>5bc560bb3554b8d705ec36e3905726f5</md5_checksum>
    </File>

    <Table_Character>
      <name>Test Instrument Table Data</name>
      <offset unit="byte">0</offset>
      <records>4</records>
      <description>This table is a Character table</description>
      <record_delimiter>Carriage-Return Line-Feed</record_delimiter>
```

Reply:
additionally, I don't think we care about the ASCII value. you should just be able to just check if the string ends with `\r\n` or just `\n`. Let me know if that makes sense? or we can setup some time tomorrow to discuss